sw/host/opentitantool/src/command/emulator.rs

```diff
@@ -0,0 +1,139 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use erased_serde::Serialize;
+use std::any::Any;
+use std::collections::HashMap;
+use std::path::PathBuf;
+use structopt::StructOpt;
+
+use opentitanlib::app::command::CommandDispatch;
+use opentitanlib::app::TransportWrapper;
+use opentitanlib::io::emu::EmuValue;
+use opentitanlib::transport::Capability;
+
+#[derive(Debug, StructOpt)]
+/// Get State of Emulator instance
+pub struct EmuGetState {}
+
+#[derive(serde::Serialize)]
+pub struct EmuGetStateResult {
+    pub state: String,
+}
+
+impl CommandDispatch for EmuGetState {
+    fn run(
+        &self,
+        _context: &dyn Any,
+        transport: &TransportWrapper,
+    ) -> Result<Option<Box<dyn Serialize>>> {
+        transport
+            .capabilities()?
+            .request(Capability::EMULATOR)
+            .ok()?;
+        let emulator = transport.emulator()?;
+        let state = emulator.get_state()?;
+        Ok(Some(Box::new(EmuGetStateResult {
+            state: state.to_string(),
+        })))
+    }
+}
+
+#[derive(Debug, StructOpt)]
+/// Start Emulator instance
+pub struct EmuStart {
+    #[structopt(
+        long,
+        help = "Reset all presistent storage (For example: flash, otp, eeprom) to factory state"
+    )]
+    pub factory_reset: bool,
+    #[structopt(long, help = "Emulator executable file name")]
+    pub emulator_exec: Option<String>,
+    #[structopt(
+        long,
+        help = "List of application names that will be provided in flash images"
+    )]
+    pub apps_list: Option<Vec<String>>,
+    #[structopt(
+        long,
+        parse(from_os_str),
+        help = "List of file paths representing Flash images"
+    )]
+    pub flash_list: Option<Vec<PathBuf>>,
+    #[structopt(
+        long,
+        parse(from_os_str),
+        help = "Path to file representing Emulator version state"
+    )]
+    pub version_init_state: Option<PathBuf>,
+    #[structopt(
+        long,
+        parse(from_os_str),
+        help = "Path to file representing PMU initial state"
+    )]
+    pub pmu_init_state: Option<PathBuf>,
+}
+
+fn pack_args<'a, T: From<T>>(
```

Comment:
T: From<T> is tautological (and, thus, redundant).

Reply:
I need to think about it.